frontend/issue-detail-side-view.tsx

```diff
@@ -0,0 +1,299 @@
+  onCloseDetail: VoidFunction;
+  onUpdateIssues: (issueUpdates: IssueUpdate[]) => void;
+  issues: Issue[];
+  rep: Replicache<M>;
+  onCreateComment: (comment: Comment) => Promise<void>;
+}
+
+export const IssueDetailSideView: React.FC<Props> = ({
+  issueId,
+  onCloseDetail,
+  issues,
+  rep,
+  onUpdateIssues,
+  onCreateComment,
+}) => {
+  const issue = issues.find((issue) => issue.id === issueId);
+
+  const [inEditMode, setEditMode] = useState(false);
+  const [titleText, setTitleText] = useState("");
+  const [descriptionText, setDescriptionText] = useState("");
+  const [commentText, setCommentText] = useState("");
+
+  const description = useSubscribe(
+    rep,
+    async (tx) => {
+      if (issueId) {
+        return (await getIssueDescription(tx, issueId)) || null;
+      }
+      return null;
+    },
+    { default: null, dependencies: [issueId] }
+  );
+
+  const comments = useSubscribe(
+    rep,
+    async (tx) => {
+      if (issueId) {
+        return (await getIssueComments(tx, issueId)) || [];
+      }
+      return [];
+    },
+    { default: [], dependencies: [issueId] }
+  );
+
+  const onEdit = () => {
+    setTitleText(issue?.title || "");
+    setDescriptionText(description || "");
+    setEditMode(true);
+  };
+
+  const onSave = () => {
+    if (issue) {
+      const descriptionUpdate =
+        descriptionText !== description
+          ? {
+              description: description || "",
+              descriptionChange: descriptionText,
+            }
+          : undefined;
+
+      onUpdateIssues([
+        {
+          issue,
+          issueChanges:
+            titleText !== issue.title
+              ? {
+                  title: titleText,
+                }
+              : {},
+          descriptionUpdate,
+        },
+      ]);
+    }
+    setEditMode(false);
+  };
+
+  const onChangeStatus = useCallback(
+    (status: Status) => {
+      issue && onUpdateIssues([{ issue, issueChanges: { status } }]);
+    },
+    [onUpdateIssues, issue]
+  );
+
+  const onChangePriority = useCallback(
+    (priority: Priority) => {
+      issue && onUpdateIssues([{ issue, issueChanges: { priority } }]);
+    },
+    [onUpdateIssues, issue]
+  );
+
+  const onClose = () => {
+    onCloseDetail();
+    setEditMode(false);
+  };
+
+  const onComment = useCallback(() => {
+    if (commentText !== "") {
+      onCreateComment(
+        {
+          id: nanoid(),
+          issueID: issue?.id as string,
+          created: Date.now(),
+          creator: "Me",
+          body: commentText,
+        } as unknown as Comment /** Needs type fixing although same code was used */
+      );
+      setCommentText("");
+    }
+  }, [onCreateComment, commentText, issue]);
+
+  return (
+    <div
+      className={classNames(
+        "bg-gray-900 top-0 bottom-0 right-0 transition-width max-w-xl z-10",
+        {
+          "w-1/3": issueId,
+          "w-0": !issueId,
+        }
+      )}
+    >
+      {!!issue && (
```

Comment:
Huge component that can be split in the next run. Furthermore rerendering needs to be checked properly for Radix components and respectively using useCallbacks. Utility functions can be moved out of this file.